qmt/data/geometry.py

```diff
@@ -151,6 +151,79 @@ def edge_coord_list(self, edge_name):
         coord_list = list(np.array(self.edges[edge_name].coords.xy).T)[:]
         return coord_list
 
+    def plot(
+        self,
+        axis_limits: Sequence[float] = [],
+        parts_to_exclude: Sequence[str] = [],
+        line_width: float = 20.0,
+        figsize: Sequence[float] = (10, 10),
+        colors: Sequence = list(mcd.XKCD_COLORS.values()),
+    ):
+        """
+        Plots the 2d geometry
+        :param axis_limits: Sequence with 4 values: xmin, ymin, xmax, ymax 
+        :param parts_to_exclude: Part/edge names that won't be plotted
+        :param line_width: Thickness of lines (only for edge lines)
+        :param figsize: Size of the figure
+        """
+        from matplotlib import pyplot as plt
+        import descartes
+
+        fig = plt.figure(figsize=figsize)
+        ax = fig.add_subplot(111)
+        pn = 0
+        for part_name, part in self.edges.items():
+            if part_name in parts_to_exclude:
+                continue
+            if len(part.coords) == 2:
+                coords = np.asarray(part.coords)
+                vec = np.asarray(coords[0]) - np.asarray(coords[1])
+                vec /= np.linalg.norm(vec)
+                perp_vec = np.array([-vec[1], vec[0]])
+                half_width = line_width / 2
+                part1 = LineString(
+                    [
+                        coords[0] + half_width * perp_vec,
+                        coords[1] + half_width * perp_vec,
+                        coords[1] - half_width * perp_vec,
+                        coords[0] - half_width * perp_vec,
+                    ]
+                )
+            else:
+                part1 = part
+            pgn = Polygon(LinearRing(part1))
+            patch = descartes.PolygonPatch(pgn, fc=colors[pn].upper(), label=part_name)
+            ax.add_patch(patch)
+
+            plt.text(
+                list(*part.representative_point().coords)[0],
+                list(*part.representative_point().coords)[1],
+                part_name,
+                ha="center",
+                va="center",
+            )
+            pn += 1
+
+        for part_name, part in self.parts.items():
+            if part_name in parts_to_exclude:
+                continue
+            patch = descartes.PolygonPatch(part, fc=colors[pn].upper(), label=part_name)
+            ax.add_patch(patch)
+            plt.text(
+                list(*part.representative_point().coords)[0],
+                list(*part.representative_point().coords)[1],
+                part_name,
+                ha="center",
+                va="center",
+                size=14,
+            )
+            pn += 1
+        if axis_limits:
+            ax.axis(axis_limits)
```

Comment:
If you return an axis - you don't need this at all - a user can do it themselves before calling plt.show()

Reply:
Returning axis now